packages/webpack/src/buildtime/modulesData.js

```diff
@@ -167,9 +168,12 @@ exports.analyzeModules = ({
     // TODO: refactor to move random hardening of the build somewhere it's easier to track.
     if (
       isAmbientAsset(module) &&
-      (module?.resourceResolveData?.context?.issuer
-        ? module.resourceResolveData.context.issuer.includes('node_modules') // resourceResolveData.context.issuer is the module that requested the asset to be present (eg. contained a `new URL(asset)`)
-        : module.resource.includes('node_modules'))
+      (module?.resourceResolveData?.context &&
+      'issuer' in module.resourceResolveData.context
+        ? // @ts-expect-error - webpack shipped a type for context that's just `object` which is useless
+          module.resourceResolveData.context.issuer.includes('node_modules') // resourceResolveData.context.issuer is the module that requested the asset to be present (eg. contained a `new URL(asset)`)
+        : typeof module.resource === 'string' &&
+          module.resource.includes('node_modules'))
```

Comment:
this whole if clause would be so much simpler if typescript would just bugger off